Userland/Services/Taskbar/TaskbarWindow.cpp

```diff
@@ -349,6 +441,28 @@ void TaskbarWindow::screen_rects_change_event(GUI::ScreenRectsChangeEvent& event
     on_screen_rects_change(event.rects(), event.main_screen_index());
 }
 
+void TaskbarWindow::config_string_did_change(String const& domain, String const& group, String const& key, String const& value)
+{
+    if (domain == "Taskbar" && group == "Appearance" && key == "Location") {
+        Gfx::Alignment new_location;
+        if (value == "Bottom")
+            new_location = Gfx::Alignment::Bottom;
+        else if (value == "Left")
+            new_location = Gfx::Alignment::Left;
+        else if (value == "Top")
+            new_location = Gfx::Alignment::Top;
+        else if (value == "Right")
+            new_location = Gfx::Alignment::Right;
+        else
+            VERIFY_NOT_REACHED();
+
+        // Kill ourselves to force a complete reflow. :^)
+        // This is currently needed due to a lot of reasons, but mainly to ensure all positions and sizes are correct.
+        if (new_location != m_location)
+            exit(0);
+    }
```

Comment:
This feels extremely hackish. Is there really no better way than dying and having `SystemServer` restart us?

Reply:
I agree fully agree although I decided to go this route since so much of the taskbar layout relies on the layout. I can update the comment to add a fixme notice.